packages/components/src/internal/components/editable/EditableGrid.tsx

```diff
@@ -858,7 +869,7 @@ export class EditableGrid extends PureComponent<EditableGridProps, EditableGridS
                     cell: inputCellFactory(
                         queryInfo,
                         editorModel,
-                        allowSelection,
+                        allowSelection_,
```

Comment:
Needed to respect the computed `allowSelection_` here to get column indices to align.